daemon/daemon/ipv4_prefix.go

```diff
@@ -0,0 +1,188 @@
+//
+// Copyright 2016 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package daemon
+
+import (
+	"encoding/json"
+	"path"
+	"strconv"
+
+	"github.com/cilium/cilium/common"
+	"github.com/cilium/cilium/common/addressing"
+)
+
+func (d *Daemon) updateIPv4Prefix(nodeIPv4 *addressing.NodeIPv4Prefix) error {
+	key := path.Join(common.IPv4PrefixKeyPath, strconv.FormatUint(uint64(nodeIPv4.ID), 10))
+	return d.kvClient.SetValue(key, nodeIPv4)
+}
+
+// gasNewSecLabelID gets and sets a New SecLabel ID.
+func (d *Daemon) gasNewIPv4Prefix(nodeIPv4 *addressing.NodeIPv4Prefix) error {
+	baseIPv4Prefix, err := d.GetMaxIPv4Prefix()
```

Comment:
We'll need the same for IPv6 as well. It's less urgent but the current method of deriving it from the IPv4 address will not always work as it's perfectly valid to have overlapping IPv4 addresses for the host itself.

At the very least, we should verify that the IPv6 node address is not conflicting.